src/resources/UsageAnalytics/Read/DataHealth/DataHealthInterfaces.ts

```diff
@@ -0,0 +1,350 @@
+     */
+    eventId: string;
+    /**
+     * The timestamp of the event.
+     */
+    timestamp: string;
+    /**
+     * The raw payload of the event.
+     */
+    eventPayload: Record<string, unknown>;
+}
+
+export interface DataHealthValidationErrorModel {
+    /**
+     * The error code.
+     */
+    code: string;
+    /**
+     * The path of the parameter causing the error.
+     */
+    path: string;
+    /**
+     * A description of the error.
+     */
+    message: string;
+}
+
+export interface DataHealthEventsListItemModel {
+    /**
+     * The source of the event.
+     */
+    source: string;
+    /**
+     * The type of the event.
+     */
+    type: string;
+    /**
+     * The searchHub of the event.
+     */
+    searchHub: string;
+    /**
+     * The ISO-8601 formatted timestamp of the event.
+     */
+    timestamp: string;
+    /**
+     * The visitId of the event.
+     */
+    visitId: string;
+    /**
+     * The clientId of the event.
+     */
+    clientId: string;
+    /**
+     * The id of the event.
+     */
+    eventId: string;
+    /**
+     * The validation errors of the event.
+     */
+    validationErrors: DataHealthValidationErrorModel[];
+}
+
+interface PaginatedResponse {
+    /**
+     * Pagination information.
+     */
+    pagination: PaginationInformationModel;
+}
+
+export interface DataHealthListEventsResponse extends PaginatedResponse {
+    /**
+     * The listing of events in the current page.
+     */
+    events: DataHealthEventsListItemModel[];
+}
+export interface DataHealthListFacetValueParams
+    extends OrganizationParamParts,
+        TimeRangeParamParts,
+        DataHealthFilterEventsParamParts {
+    /**
+     * The facet for which the values should be listed.
+     */
+    facet: string;
+}
+
+export interface DataHealthFacetValue<TValue = string> {
+    /**
+     * The value of the facet.
+     */
+    value: TValue;
+    /**
+     * The count of records that matched the facet in the current range.
+     */
+    count: number;
+}
+
+export interface DataHealthListFacetsResponse {
+    /**
+     * The values for this facet.
+     */
+    values: any[];
+}
+
+export interface DataHealthGetOverviewParams extends OrganizationParamParts, TimeRangeParamParts {
+    /**
+     * The category to filter by when getting the overview.
+     */
+    category?: string;
+}
+
+export interface DataHealthGetOverviewResponse {
+    /**
+     * The overall health score for the specified category(ies)
+     */
+    score: number;
+    /**
+     * The breakdown of overall validation failures by severity.
+     */
+    severityBreakdown: Record<Severity, number>;
+    /**
+     * The total number of entity validated.
+     */
+    totalValidations: number;
+    /**
+     * The total number of entity that failed validation.
+     */
+    failedValidations: number;
+}
+
+export type Severity = 'CRITICAL' | 'HIGH' | 'MEDIUM' | 'LOW';
+Object.freeze<Severity[]>(['CRITICAL', 'HIGH', 'MEDIUM', 'LOW']);
```

Comment:
This is a useless statement, it creates a frozen array that is not assigned to anything.
Did you mean to port this:
```suggestion
export const SeverityConstant = Object.freeze<Severity[]>(['CRITICAL', 'HIGH', 'MEDIUM', 'LOW']);
```
If not, remove this.